modules/standard/Collections/Mutable/HashMap.enc

```diff
@@ -78,16 +78,16 @@ local class HashMapIterator[k : Hashable + Eq[k], v]
     Returns Nothing if the end of the HashMap
     has been reached.
   -}
-  def next() : Maybe[Entry[k,v]]
+  def next() : Entry[k,v]
     val current = this.current_node
 
     if this.has_next() then
       this.current_node = this.get_next_node()
     end
 
     match current with
-      case Just(n) => Just(n.value)
-      case Nothing => Nothing
+      case Just(n) => n.value
+      case Nothing => abort("No next element")
```

Comment:
Does this mean that a user cannot do `iterator.next().next()` she knows that exact number of items? If the iterator doesn't have any elements, doing `iterator.next()` crashes, doesn't it?